### PR TITLE
When measuring violations with SonarQube as source, add the selected …

### LINKED ISSUE #4449: Make QT use the same severities for a link as selected in the source
I'm always frustrated when I click the link to see violations in Sonar and I end up at a page showing all violations.

I would like QT to use a link with the violation severities selected that are used in the source config.

image

image

### PR DESCRIPTION
**Pull request by fniessink**
…issue types (bug, vulnerability, and/or code smell) and severities (info, minor, major, critical, and/or blocker) to the landing URL so that the user sees only the relevant violations when navigating to SonarQube. Closes #4449.

### CODE REVIEW

File: components/collector/src/source_collectors/sonarqube/security_warnings.py [hunk 1]
```diff
@@ -19,19 +19,17 @@ async def _landing_url(self, responses: SourceResponses) -> URL:
...
         component = self._parameter("component")
         branch = self._parameter("branch")
+        common_url_parameters, extra_url_parameters = f"?id={component}&branch={branch}", ""
         if "vulnerability" in security_types and "security_hotspot" in security_types:
-            landing_url = f"{base_landing_url}/dashboard?id={component}&branch={branch}"
+            landing_path = "dashboard"
         elif "vulnerability" in security_types:
-            landing_url = f"{base_landing_url}/project/issues?id={component}&resolved=false&branch={branch}"
+            landing_path = "project/issues"
+            extra_url_parameters = f"{self._query_parameter('severities')}&resolved=false&types=VULNERABILITY"
```
Comment (wkoot) [inline components/collector/src/source_collectors/sonarqube/security_warnings.py:28]: should hardcoded `types=VULNERABILITY` not be combined with `_query_parameter(self.types_parameter)`?

Comment (wkoot) [reply components/collector/src/source_collectors/sonarqube/security_warnings.py:28]: or is that automagically done by order of combining in `sonarqube/violations.py`.. hmm probably good to add a comment

File: components/collector/tests/source_collectors/sonarqube/test_violations.py [hunk 2]
```diff
@@ -54,4 +54,24 @@ async def test_violations(self):
...
+    async def test_violations(self):
+        """Test that the number of violations is returned."""
+        response = await self.collect(get_request_json_return_value=self.json)
+        self.assert_measurement(
+            response, value="2", entities=self.expected_entities, landing_url=self.issues_landing_url
+        )
+
+    async def test_major_violations(self):
+        """Test that the number of major violations is returned."""
+        self.set_source_parameter("severities", ["major"])
```
Comment (wkoot) [inline components/collector/tests/source_collectors/sonarqube/test_violations.py:67]: perhaps good to set two severities, to also verify that the comma sep urlencoded `%2C` works

**Review by wkoot — CHANGES_REQUESTED** [review]

File: components/collector/src/source_collectors/sonarqube/security_warnings.py (hunk 1 above)
Comment (fniessink) [reply components/collector/src/source_collectors/sonarqube/security_warnings.py:28]: Using `_query_parameter(self.types_parameter)` would result in `&security_types=...` but we need `&types=...`. I'll add a comment.

**Pull request merged**